src/main/java/io/r2dbc/h2/H2Statement.java

```diff
@@ -146,9 +149,11 @@ private H2Statement addIndex(int index, Object value) {
                         ResultWithGeneratedKeys result = client.update(command, generatedColumns);
                         CommandUtil.clearForReuse(command);
                         if (GeneratedKeysMode.valueOf(generatedColumns) == GeneratedKeysMode.NONE) {
-                            return H2Result.toResult(codecs, result.getUpdateCount());
+                            int updatedCountInt = Long.valueOf(result.getUpdateCount()).intValue();
```

Comment:
1. `(int) result.getUpdateCount()` does exactly the same, but without allocation of `java.lang.Long`.
2. H2 actually can return values larger than `Integer.MAX_VALUE` from some operations, for example, from `TRUNCATE TABLE`, so it would be better to handle it in some better way.